src/content/content_engine.rs

```diff
@@ -258,30 +248,16 @@ where
                 // need to both live in the handlebars registry under distinct
                 // names).
                 let template_name = content.relative_path;
-                let mut contents = content.file;
                 if handlebars_registry.has_template(&template_name) {
                     return Err(ContentLoadingError::Bug(format!(
                         "More than one handlebars template has the name '{}'.",
                         template_name,
                     )));
                 }
                 handlebars_registry
-                    .register_template_source(&template_name, &mut contents)
-                    .map_err(|template_render_error| match template_render_error {
-                        handlebars::TemplateFileError::TemplateError(source) => {
-                            ContentLoadingError::TemplateParseError(TemplateParseError { source })
-                        }
-                        handlebars::TemplateFileError::IOError(source, original_name) => {
-                            // Handlebars-rust will use an empty string when the
-                            // error does not correspond to a specific path.
-                            let name = if original_name.is_empty() {
-                                None
-                            } else {
-                                Some(original_name)
-                            };
-                            ContentLoadingError::IOError { source, name }
-                        }
-                    })?;
+                    .register_template_file(&template_name, content.absolute_path)
```

Comment:
To avoid handlebars-rust needing to reopen the file, I could instead read `content.file` into a `String` and use `register_template_string` here. That might be more efficient, but I haven't benchmarked the two approaches against each other (since this only impacts startup time it isn't super important).

I went with the `register_template_file` approach because it's simpler and because it opens up the option of using [`dev_mode`](https://github.com/sunng87/handlebars-rust#auto-reload-in-dev-mode), which could be useful until #9 is fully addressed.